src/sentry/ratelimits/redis.py

```diff
@@ -106,8 +106,11 @@ def is_limited_with_value(
         # Reset Time = next time bucket's start time
         reset_time = _bucket_start_time(_time_bucket(request_time, window) + 1, window)
         try:
-            result = self.client.incr(redis_key)
-            self.client.expire(redis_key, expiration)
+            pipe = self.client.pipeline()
+            pipe.multi()
```

Comment:
_MULTI_ makes it a transaction, if the goal of this PR is only to improve the performance, then pipeline w/o a transaction will be enough.

Reply:
I've updated the implementation a little bit, but I left the multi since it provides transactions and reverts expiration if increment fails.